src/main/scala/is/hail/io/RowStore.scala

```diff
@@ -744,19 +746,20 @@ class RichRDDRegionValue(val rdd: RDD[RegionValue]) extends AnyVal {
     val nPartitions = rdd.getNumPartitions
     val d = digitsNeeded(nPartitions)
 
-    val partFiles = Array.tabulate[String](nPartitions) { i => partFile(d, i) }
-
     val fullRowType = t.rvRowType
     val rowsRVType = t.rowType
     val localEntriesIndex = t.entriesIdx
 
     val entriesRVType = TStruct(
       MatrixType.entriesIdentifier -> TArray(t.entryType))
 
-    val partitionCounts = rdd.mapPartitionsWithIndex { case (i, it) =>
+    val partFilePartitionCounts = rdd.mapPartitionsWithIndex { case (i, it) =>
       val hConf = sHConfBc.value.value
 
-      val f = partFile(d, i)
+      val rng = new java.security.SecureRandom()
+      val fileUUID = new java.util.UUID(rng.nextLong(), rng.nextLong())
+      val ctx = TaskContext.get
+      val f = partFile(d, i) + s"-${ ctx.stageId() }-${ ctx.partitionId() }-${ ctx.attemptNumber() }-${ fileUUID }"
```

Comment:
This complex string is already used in three places, so I think it'd make sense to abstract the definition.